### PR TITLE
Improved chunking index speeds

### PR DESCRIPTION
**Pull request by lettergram**
Shuffling indices for sampling is one of the slowest aspects of the library, see below for a 52Mb file with 2.7 million records. Shuffling is about 18% of the runtime. The proposal here reduces the overall runtime by ~6 seconds or 10%. 

I believe @JGSweets had something similar previously. Generally, python's `random` is faster. However, given the vectorized nature here, I recommend utilizing numpy. 

```
60479794 function calls (60295604 primitive calls) in 53.637 seconds

   Ordered by: internal time
   List reduced from 9957 to 50 due to restriction <50>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
      126    5.188    0.041    6.512    0.052 numerical_column_stats.py:239(_total_histogram_bin_variance)
       12    4.128    0.344    9.595    0.800 utils.py:50(shuffle_in_chunks)
        6    3.089    0.515    4.544    0.757 float_column_profile.py:98(_get_float_precision)
    17982    2.883    0.000    3.055    0.000 numerical_column_stats.py:386(_get_percentile)
  3343290    2.737    0.000    6.006    0.000 base_column_profilers.py:47(_combine_unique_sets)
```

I then did a test with the current chunking function with 10 million records and a chunk_size of 10 thousand:
```
54578261 function calls in 23.893 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1001   11.084    0.011   23.772    0.024 shuffle_in_chunks.py:17(shuffle_in_chunks)
 10000000    5.136    0.000   11.887    0.000 random.py:291(randrange)
 10000000    4.714    0.000    6.751    0.000 random.py:238(_randbelow_with_getrandbits)
 14575255    1.323    0.000    1.323    0.000 {method 'getrandbits' of '_random.Random' objects}
```

The method proposed here (10 million records, chunk_size 10 thousand):
```
10049030 function calls in 13.806 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1001   12.514    0.013   13.687    0.014 shuffle_in_chunks.py:52(shuffle_in_chunks_v2)
 10000001    0.632    0.000    0.632    0.000 shuffle_in_chunks.py:14(__missing__)
     4000    0.339    0.000    0.339    0.000 {method 'items' of 'dict' objects}
     1000    0.154    0.000    0.535    0.001 {method 'integers' of 'numpy.random._generator.Generator' objects}
        1    0.119    0.119   13.806   13.806 shuffle_in_chunks.py:177(test_chunk_v2)
     4000    0.012    0.000    0.012    0.000 {method 'reduce' of 'numpy.ufunc' objects}
     4000    0.006    0.000    0.363    0.000 fromnumeric.py:70(_wrapreduction)
     4001    0.006    0.000    0.372    0.000 {built-in method numpy.core._multiarray_umath.implement_array_function}
     1000    0.005    0.000    0.005    0.000 {built-in method numpy.zeros}
```

### CODE REVIEW

File: dataprofiler/profilers/utils.py
Comment (AnhTruong) [inline dataprofiler/profilers/utils.py:87]: should be `random_list[j]` ?

File: dataprofiler/profilers/utils.py
Comment (AnhTruong) [inline dataprofiler/profilers/utils.py:79]: can it just be `lower_bound_list = np.array(range(j, j + true_chunk_size))`

**Review by AnhTruong — COMMENTED** [review]
the changes generally looks good. Waiting for all tests to pass before I approve.

File: dataprofiler/profilers/utils.py
Comment (lettergram) [reply dataprofiler/profilers/utils.py:87]: No, this is correct, "j" is globally incrementing. "Count" is incrementing only for this chunk. I only generate the random numbers for this chunk, so if we use global "j" it'll fail.

File: dataprofiler/profilers/utils.py
Comment (lettergram) [reply dataprofiler/profilers/utils.py:79]: I believe you're correct.

Comment (lettergram) [reply dataprofiler/profilers/utils.py:79]: Current`` method:
```
10049030 function calls in 11.812 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1001   10.818    0.011   11.693    0.012 shuffle_in_chunks.py:52(shuffle_in_chunks_v2)
 10000001    0.481    0.000    0.481    0.000 shuffle_in_chunks.py:14(__missing__)
     4000    0.225    0.000    0.225    0.000 {method 'items' of 'dict' objects}
     1000    0.136    0.000    0.389    0.000 {method 'integers' of 'numpy.random._generator.Generator' objects}
        1    0.119    0.119   11.812   11.812 shuffle_in_chunks.py:178(test_chunk_v2)

```

Your proposal:
```
10049029 function calls in 11.453 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1001    9.681    0.010   11.336    0.011 shuffle_in_chunks.py:52(shuffle_in_chunks_v2)
     1000    0.797    0.001    0.797    0.001 {built-in method numpy.array}
 10000000    0.488    0.000    0.488    0.000 shuffle_in_chunks.py:14(__missing__)
     4000    0.227    0.000    0.227    0.000 {method 'items' of 'dict' objects}
        1    0.116    0.116   11.453   11.453 shuffle_in_chunks.py:178(test_chunk_v2)
```

**Review by JGSweets — APPROVED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by AnhTruong — APPROVED** [review]

**Pull request merged**
